tools/mdfileconverter/md_converter.xml

```diff
@@ -0,0 +1,124 @@
+<tool id="md_converter" name="MDTraj file converter" version="1.9.1">
+    <macros>
+        <xml name='traj' token_format='format'>
+            <when value="@FORMAT@">
+                <param argument="input_file" type="data" format='@FORMAT@' label="@FORMAT@ input file for conversion"/>
+                <param argument="output_format" type="select" label="Output">
+                    <option value="trr">TRR file</option>
+                    <option value="xtc">XTC file</option>
+                    <option value="dcd">DCD file</option>
+                </param>
+            </when>
+        </xml>
+        <xml name='str' token_format='format'>
+            <when value="@FORMAT@">
+                <param argument="input_file" type="data" format='@FORMAT@' label="@FORMAT@ input file for conversion"/>
+                <param argument="output_format" type="select" label="Output">
+                    <option value="gro">GRO file</option>
+                    <option value="pdb">PDB file</option>
+                </param>
+            </when>
+        </xml>
+    </macros>
+
+    <description>- interconvert between MD trajectory or structure file formats.</description>
+
+    <requirements>
+        <requirement type="package" version="1.9.1">mdtraj</requirement>
+        <requirement type="package" version="2018.2">gromacs</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        #if $input.input_format == 'gro' or $input.input_format == 'pdb':
```

Comment:
You can use `$input_file.ext` to get the format. With this you could do a single input field without the conditional but wit `format="pdb,trr,xtc"`

Reply:
I see ... leave it like that, sorry for the noise.